Macos Homebrew fatal error: label followed by a declaration is a C23 extension

### Is there an existing issue for this?

- [x] I checked and did not find my issue in the already reported ones

### Describe the bug

release-5.5.0-149-gb6d2433ab9

...
...darktable/src/develop/masks/gradient.c:1489:7: fatal error: label followed by a declaration is a C23 extension [-Wc23-extensions]
 1489 |       float curvature = gradient
      |       ^
1 error generated.




### Steps to reproduce

on MacOs Tahoe and master:

1) cd darktable/packaging/macosx

2) ./1_install_hb_dependencies.sh
3) ./2_build_hb_darktable_default.sh

### Expected behavior

compile w/o error

### Logfile | Screenshot | Screencast

_No response_

### Commit

_No response_

### Where did you obtain darktable from?

darktable.org / GitHub release

### darktable version

release-5.5.0-149-gb6d2433ab9

### What OS are you using?

Mac

### What is the version of your OS?

Tahoe 26.2

### Describe your system

_No response_

### Are you using OpenCL GPU in darktable?

None

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

Linux build OK

--- a/src/develop/masks/gradient.c
+++ b/src/develop/masks/gradient.c
@@ -1486,54 +1486,54 @@ static void _gradient_modify_property(dt_masks_form_t *const form,
   switch(prop)
   {
     case DT_MASKS_PROPERTY_CURVATURE:
-    {
-      float curvature = gradient
-        ? gradient->curvature
-        : dt_conf_get_float(DT_MASKS_CONF(form->type, gradient, curvature));
-      curvature = CLAMP(curvature + 2.0f * (new_val - old_val), -2.0f, 2.0f);
+      {
+        float curvature = gradient
+          ? gradient->curvature
+          : dt_conf_get_float(DT_MASKS_CONF(form->type, gradient, curvature));
+        curvature = CLAMP(curvature + 2.0f * (new_val - old_val), -2.0f, 2.0f);
 
-      if(gradient) gradient->curvature = curvature;
-      dt_conf_set_float(DT_MASKS_CONF(form->type, gradient, curvature), curvature);
+        if(gradient) gradient->curvature = curvature;
+        dt_conf_set_float(DT_MASKS_CONF(form->type, gradient, curvature), curvature);
 
-      *sum += curvature * 0.5;
-      *max = fminf(*max,  1.0f - 0.5 * curvature);
-      *min = fmaxf(*min, -1.0f - 0.5 * curvature);
-      ++*count;
+        *sum += curvature * 0.5;
+        *max = fminf(*max,  1.0f - 0.5 * curvature);
+        *min = fmaxf(*min, -1.0f - 0.5 * curvature);
+        ++*count;
+      }
       break;
-    }
 
     case DT_MASKS_PROPERTY_COMPRESSION:
-    {
-      const float ratio = (!old_val || !new_val) ? 1.0f : new_val / old_val;
-      float compression = gradient
-        ? gradient->compression
-        : dt_conf_get_float(DT_MASKS_CONF(form->type, gradient, compression));
-      compression = CLAMP(compression * ratio, 0.001f, 1.0f);
+      {
+        const float ratio = (!old_val || !new_val) ? 1.0f : new_val / old_val;
+        float compression = gradient
+          ? gradient->compression
+          : dt_conf_get_float(DT_MASKS_CONF(form->type, gradient, compression));
+        compression = CLAMP(compression * ratio, 0.001f, 1.0f);
 
-      if(gradient) gradient->compression = compression;
-      dt_conf_set_float(DT_MASKS_CONF(form->type, gradient, compression), compression);
+        if(gradient) gradient->compression = compression;
+        dt_conf_set_float(DT_MASKS_CONF(form->type, gradient, compression), compression);
 
-      *sum += compression;
-      *max = fminf(*max, 1.0f / compression);
-      *min = fmaxf(*min, 0.0005f / compression);
-      ++*count;
+        *sum += compression;
+        *max = fminf(*max, 1.0f / compression);
+        *min = fmaxf(*min, 0.0005f / compression);
+        ++*count;
+      }
       break;
-    }
 
     case DT_MASKS_PROPERTY_ROTATION:
-    {
-      float rotation = gradient
-        ? gradient->rotation
-        : dt_conf_get_float(DT_MASKS_CONF(form->type, gradient, rotation));
-      rotation = fmodf(rotation - new_val + old_val + 360.0f, 360.0f);
+      {
+        float rotation = gradient
+          ? gradient->rotation
+          : dt_conf_get_float(DT_MASKS_CONF(form->type, gradient, rotation));
+        rotation = fmodf(rotation - new_val + old_val + 360.0f, 360.0f);
 
-      if(gradient) gradient->rotation = rotation;
-      dt_conf_set_float(DT_MASKS_CONF(form->type, gradient, rotation), rotation);
+        if(gradient) gradient->rotation = rotation;
+        dt_conf_set_float(DT_MASKS_CONF(form->type, gradient, rotation), rotation);
 
-      *sum += 360.0f - rotation;
-      ++*count;
+        *sum += 360.0f - rotation;
+        ++*count;
+      }
       break;
-    }
 
     default: {}
   }

